tests/tox_env/python/virtual_env/package/test_python_package_util.py

```diff
@@ -67,5 +67,13 @@ def test_loads_deps_recursive_extras() -> None:
 
 def test_load_dependency_requirement_or_extras() -> None:
     requires = [Requirement('filelock<4.0.0,>=3.9.0; extra == "extras1" or extra == "extras2"')]
-    result = dependencies_with_extras(requires, {"extras1"}, "")
-    assert [str(r) for r in result] == ["filelock<4.0.0,>=3.9.0"]
+    for extras in ["extras1", "extras2"]:
+        result = dependencies_with_extras(requires, {extras}, "")
+        assert [str(r) for r in result] == ["filelock<4.0.0,>=3.9.0"]
+
+
+def test_load_dependency_requirement_many_or_extras() -> None:
+    requires = [Requirement('filelock<4.0.0,>=3.9.0; extra == "extras1" or extra == "extras2" or extra == "extras3"')]
```

Comment:
While we're at it, can we also handle and expression? 🤔 we should also add in the test a python version marker to make sure that's kept and we only strip extras

Reply:
Yeah I was thinking there are probably all kinds of corner cases / nasty expressions we can come up with to break this... I'll add some more tests

Do we have an example of a realistic "and" expression we want to support?  Or is it purely hypothetical/synthetic?

Reply:
I haven't seen it but is possible 🤔